packages/pg/lib/connection-parameters.js

```diff
@@ -26,7 +27,35 @@ var readSSLConfigFromEnvironment = function () {
     case 'require':
     case 'verify-ca':
     case 'verify-full':
-      return true
+      var caFileName = val('sslrootcert', {})
+      if (!caFileName) {
+        throw new Error('Postgres SSL connection requested using PGSSLMODE environment variable, but no PGSSLROOTCERT environment variable defined')
+      }
+
+      var crtFileName = val('sslcert', {})
+      if (!crtFileName) {
+        throw new Error('Postgres SSL connection requested using PGSSLMODE environment variable, but no PGSSLCERT environment variable defined')
+      }
+
+      var keyFileName = val('sslkey', {})
+      if (!keyFileName) {
+        throw new Error('Postgres SSL connection requested using PGSSLMODE environment variable, but no PGSSLKEY environment variable defined')
+      }
+
+      var result = {
+        rejectUnauthorized: false,
+        ca: fs.readFileSync(caFileName).toString(),
+        cert: fs.readFileSync(crtFileName).toString()
+      }
+
+      // "hiding" the private key so it doesn't show up in stack traces
+      // or if the client is console.logged
+      Object.defineProperty(result, 'key', {
+        enumerable: false,
+        value: fs.readFileSync(keyFileName).toString()
```

Comment:
please don't use blocking calls - this lib is supposed to be 100% non-blocking.

Reply:
Understood.

Reply:
BTW I already see some other sync file reads like this one: https://github.com/brianc/node-postgres/blob/master/packages/pg-connection-string/index.js#L76

I may fix it as well if you prefer.

Reply:
I tried to get rid of the synchronous `fs.readFileSync()` by using `fs.promises.readFile()`; however, this makes the whole constructor of the Client `async` as well, which is a significant breaking change and tests are failing as well.

Since there was already `fs.readFileSync()` in the code, as I mentioned in the previous comment, and since these bugs of TLS support are blocking me (and probably many other people), I would like to ask you to approve them as synchronous for the moment. Anyway, this is only a read from a (probably) local file system, this is not a client-server communication, so I guess it is worth it.

As an improvement for the future, I suggest not resolving the `ssl*` settings during connection string parsing and environment variable resolving, which are done in the constructor. They can be deferred to the `connect()` phase, which is already asynchronous. I think this is also the behavior of `psql` and `libpq`; they only get the NAMES of the files, not their content. But this change is a breaking change since existing config options are the TLS file **contents** themselves, not just the TLS file names.